### PR TITLE
TMA-956 - Update process.rb to fix regression from TMA-832

### PR DESCRIPTION
**Pull request by Hahihula**
based on vladimir-pachnik:patch-2

### CODE REVIEW

File: lib/gooddata/models/process.rb [hunk 1]
```diff
@@ -35,6 +35,7 @@ def [](id, options = {})
             c.create(Process, process_data, project: project)
           end
         elsif id == :all
+          c ||= GoodData.client # c may not defined if project is not defined
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has the variable name 'c' (UncommunicativeVariableName)

File: spec/integration/models/process_spec.rb [hunk 2]
```diff
@@ -11,12 +11,23 @@
...
   after(:all) do
     @rest_client.disconnect
   end
 
+  def create_process
+    puts "Using project: #{GoodData.project.pid} to create process"
+    GoodData::Process.deploy(@archive_location, @options.merge(name: 'Test process GRAPH'))
+  end
+
+  def destroy_process(process)
```
Comment (yenkins) [inline spec/integration/models/process_spec.rb]: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/345/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1188/ : SUCCESS in 1m 14s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1458/ : SUCCESS in 1m 03s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/346/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1190/ : SUCCESS in 1m 06s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1460/ : SUCCESS in 1m 00s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/347/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1191/ : SUCCESS in 1m 07s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1461/ : SUCCESS in 1m 00s

File: spec/integration/models/process_spec.rb [hunk 3]
```diff
@@ -11,12 +11,22 @@
...
 
   after(:all) do
     @rest_client.disconnect
   end
 
+  def create_process
+    GoodData::Process.deploy(@archive_location, @options.merge(name: 'Test process GRAPH'))
+  end
+
+  def destroy_process(process)
```
Comment (yenkins) [inline spec/integration/models/process_spec.rb]: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/348/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1192/ : SUCCESS in 1m 03s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1462/ : SUCCESS in 1m 02s

File: lib/gooddata/models/process.rb [hunk 4]
```diff
@@ -24,9 +24,10 @@ class Process < Rest::Resource
     alias_method :to_hash, :data
 
     class << self
-      def [](id, options = {})
+      def [](id, options = { :client => GoodData.connection, :project => GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has approx 20 statements (TooManyStatements)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/349/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1195/ : SUCCESS in 2m 17s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1465/ : SUCCESS in 1m 00s

File: lib/gooddata/models/process.rb [hunk 5]
```diff
@@ -24,9 +24,9 @@ class Process < Rest::Resource
     alias_method :to_hash, :data
 
     class << self
-      def [](id, options = {})
-        project = options[:project]
-        c = options[:client] || (project && project.client)
+      def [](id, options = { :client => GoodData.connection, :project => GoodData.project })
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has approx 19 statements (TooManyStatements)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/350/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1198/ : SUCCESS in 2m 24s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1468/ : SUCCESS in 2m 18s

File: lib/gooddata/models/process.rb [hunk 6]
```diff
@@ -24,32 +24,33 @@ class Process < Rest::Resource
...
         project = options[:project]
-        c = options[:client] || (project && project.client)
+        client = options[:client] || (project && project.client)
+        fail 'Client has to be specified in options' unless c
 
         if id == :all && project
           uri = "/gdc/projects/#{project.pid}/dataload/processes"
-          data = c.get(uri)
+          data = client.get(uri)
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Calls 'client.get(uri)' 3 times (DuplicateMethodCall)

File: lib/gooddata/models/process.rb [hunk 7]
```diff
@@ -24,32 +24,33 @@ class Process < Rest::Resource
...
+            client.create(Process, process_data, project: project)
           end
         elsif id == :all
-          uri = "/gdc/account/profile/#{c.user.obj_id}/dataload/processes"
-          data = c.get(uri)
+          uri = "/gdc/account/profile/#{client.user.obj_id}/dataload/processes"
+          data = client.get(uri)
           pids = data['processes']['items'].map { |process_data| process_data['process']['links']['self'].match(%r{/gdc/projects/(\w*)/})[1] }.uniq
-          projects_lookup = pids.pmap { |pid| c.projects(pid) }.reduce({}) do |a, e|
+          projects_lookup = pids.pmap { |pid| client.projects(pid) }.reduce({}) do |a, e|
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has the variable name 'a' (UncommunicativeVariableName)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Has the variable name 'e' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/353/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1200/ : SUCCESS in 1m 08s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1470/ : SUCCESS in 1m 01s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/355/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1202/ : SUCCESS in 1m 01s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1472/ : SUCCESS in 58s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/99/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1203/ : SUCCESS in 55s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1473/ : SUCCESS in 2m 24s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1463/ : FAILURE in 10m 36s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/35/ : FAILURE in 20m 34s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/361/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1207/ : SUCCESS in 2m 37s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1477/ : SUCCESS in 2m 32s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/102/ : SUCCESS in 19s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1209/ : SUCCESS in 1m 04s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1479/ : SUCCESS in 1m 00s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1465/ : FAILURE in 15m 02s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/37/ : ABORTED in 2h 00m 01s

**Comment by Hahihula** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/364/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1467/ : FAILURE in 14m 37s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/39/ : FAILURE in 32m 59s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/365/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1210/ : SUCCESS in 2m 27s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1480/ : SUCCESS in 3m 54s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/103/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1211/ : SUCCESS in 2m 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1481/ : SUCCESS in 2m 32s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1468/ : SUCCESS in 16m 05s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/40/ : FAILURE in 41m 03s

**Comment by Hahihula** [comment]
extended test - testing

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/366/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1469/ : SUCCESS in 13m 28s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/41/ : FAILURE in 32m 15s

**Comment by Hahihula** [comment]
extended test - staging

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/367/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1470/ : FAILURE in 10m 51s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/42/ : FAILURE in 32m 10s

**Comment by Hahihula** [comment]
The gooddata-ruby-lcm-integration-tests-zuul is failing because of a combination of broken staging and issue TMA-1001 but passing locally if run against staging1

**Pull request merged**